netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java

```diff
@@ -166,15 +166,12 @@ public ClientStream newStream(MethodDescriptor<?, ?> method, Metadata.Headers he
     AsciiString defaultPath = new AsciiString("/" + method.getName());
     Http2Headers http2Headers = Utils.convertClientHeaders(headers, ssl, defaultPath, authority);
 
-    // Write the command requesting the creation of the stream.
-    CreateStreamCommand msg = new CreateStreamCommand(http2Headers, stream);
-    if (method.getType().clientSendsOneMessage()) {
-      // No need to flush yet if the client is required to send at least one message
-      channel.write(msg);
-    } else {
-      channel.writeAndFlush(msg);
-    }
-
+    // Write the command requesting the creation of the stream. No need to flush if the
+    // client is required to send at least one message.
```

Comment:
I'm still not sure I understand why we would ever want to flush here. Why does the number of messages matter?